src/pods/canvas/canvas.business.ts

```diff
@@ -73,3 +79,75 @@ export const calculateRelationXCoordinate = (
   xOrigin: calculateRelationXCoordinateOrigin(tableOrigin, tableDestination),
   xDestination: calculateRelationXCoordinateEnd(tableOrigin, tableDestination),
 });
+
+const seekField = (
+  fieldId: GUID,
+  YPosition: number,
+  fields: FieldVm[]
+): seekResult => {
+  let found = false;
+  let parentCollapsed = false;
+  let newYPosition = YPosition;
+
+  for (let i = 0; i < fields.length && !found; i++) {
+    const field = fields[i];
+    if (field.id === fieldId) {
+      found = true;
+      return {
+        found,
+        parentCollapsed,
+        YPosition: newYPosition,
+      };
+    } else {
+      if (!parentCollapsed) {
+        newYPosition += ROW_HEIGHT;
+      }
```

Comment:
Let's reduce the cyclomatic complexity (Hadouken)

![image](https://github.com/Lemoncode/mongo-modeler/assets/1457912/f43b9aff-3fad-4776-838a-9093f12e127b)

Let's create a helped method:

```ts
const addFieldRowHeight = (YPosition, parentCollapsed) => (!parentColapsed) newYPosition + ROW_HEIGHT : newYPosition;
````

Then in this code:

```diff
   } else {
+    newYPosition = addFieldRowHeight(newYPosition, parentCollapsed);
-      if (!parentCollapsed) {
-        newYPosition += ROW_HEIGHT;
-      }
```

Reply:
It's fixed